scripts/segmentation_node

```diff
@@ -92,29 +130,46 @@ class SegmentationNode:
             self.counter += 1
         now  = rospy.Time.now()
         rospy.loginfo("New callback time %i %i, img_time: %i %i", now.secs, now.nsecs, img_msg.header.stamp.secs, img_msg.header.stamp.nsecs )
-        #rospy.loginfo("New callback time %i %i, img_time: %i %i", now.secs, now.nsecs, img_msg.header.stamp.secs, img_msg.header.stamp.nsecs )
         original_img = self.bridge.imgmsg_to_cv2(img_msg , desired_encoding="rgb8")
         #original_img = cv2.cvtColor(original_img, cv2.COLOR_BGR2RGB)
+        s_t = time.time()
         labeled_img, distribution = self.infer(original_img)
+        e_t = time.time()
+        print(f"Prediction time: {e_t - s_t} s")
         self.publish_label_and_distribution(labeled_img, distribution, img_msg.header)
         
     def tf_init(self):
-        print("open graph..")
-        with tf.gfile.GFile(self.network_config.path, 'rb') as f:
-            self.graph_def = tf.GraphDef()
-            self.graph_def.ParseFromString(f.read())
-        print("open graph complete")
+        # print("open graph..")
+        # with tf.gfile.GFile(self.network_config.path, 'rb') as f:
+        #     self.graph_def = tf.GraphDef()
+        #     self.graph_def.ParseFromString(f.read())
+        # print("open graph complete")
+
+        # config = tf.ConfigProto()
+        # config.gpu_options.allow_growth = True
+        # self.sess = tf.Session(config=config)
+        # self.y, = tf.import_graph_def(self.graph_def, return_elements=[self.network_config.label_output_tensor], name='')
+        # self.G = tf.get_default_graph()
+        # self.x = self.G.get_tensor_by_name(self.network_config.image_input_tensor)
+        
+        # self.dist = self.G.get_tensor_by_name(self.network_config.distribution_output_tensor)
+        # tf.global_variables_initializer().run(session=self.sess)
+        # print("Tf init finish")
+
+        print("Initializing the neural network...")
 
-        config = tf.ConfigProto()
-        config.gpu_options.allow_growth = True
-        self.sess = tf.Session(config=config)
-        self.y, = tf.import_graph_def(self.graph_def, return_elements=[self.network_config.label_output_tensor], name='')
-        self.G = tf.get_default_graph()
-        self.x = self.G.get_tensor_by_name(self.network_config.image_input_tensor)
+        self.model = sm.Unet(self.backbone, encoder_weights='imagenet', classes=21, encoder_freeze=True)
+        self.model.load_weights(self.network_config.path)
         
-        self.dist = self.G.get_tensor_by_name(self.network_config.distribution_output_tensor)
-        tf.global_variables_initializer().run(session=self.sess)
-        print("Tf init finish")
+        # Predict some random image to load the necessary libraries
+        x = cv2.imread("/home/yemika/Mikael/Work/ODRS/SegMap/network/MobileNetSeg/data/train/images/image4914.png", cv2.IMREAD_COLOR)          # read the image as BGR
```

Comment:
same problem as in the "cropping" branch, this cannot work.
Remove this whole block it's useless ( line 165 - 172 )

Reply:
You can remove it, I think I explained at some point why I did it, and it's not really mandatory.

Reply:
I remember that, but you cannot commit an absolute path that has your name in it, it will never work on other computers

Reply:
I see. Will take it into account